org.lflang/src/org/lflang/ast/AstTransformation.java

```diff
@@ -0,0 +1,16 @@
+package org.lflang.ast;
+
+import java.util.List;
+
+import org.lflang.lf.Reactor;
+
+/**
+ * Interface for AST Transfomations
+ */
+public interface AstTransformation {
+
+    /**
+     * Apply the AST transformation to all given reactors.
+     */
+    void applyTransformation(List<Reactor> reactors);
```

Comment:
I would change this to take in the entire resource so that the interface is more general.

Reply:
Yes, my first intuition was also to pass the model and/or the resource. However, it appears that there is some intelligence built into generating this list of reactors (this is handled by the instance graph). We would need to duplicate that code or refactor it to make it reusable, but I wanted to avoid spending too much time on this. Also I am not sure which information other transformations would need. I suggest to revisit this as part of fixing https://github.com/lf-lang/lingua-franca/issues/1510